crates/bevy_ecs/src/query/filter.rs

```diff
@@ -661,3 +661,109 @@ macro_rules! impl_archetype_filter_tuple {
 }
 
 all_tuples!(impl_archetype_filter_tuple, 0, 15, F);
+
+/// This macro implements `WorldQuery` and `ReadOnlyWorldQuery` for a filter
+/// so that `Filter<(A, B, C)>` acts as a shortcut for `(Filter<A>, Filter<B>, Filter<C>)`.
+macro_rules! impl_filter_tuple_to_tuple_filter {
```

Comment:
This macro and the two invocations adds yet another set of tuple pyramids, which can add a lot of additional code to compile. Can you check the difference in compile time for `bevy_ecs` before and after this change?

Reply:
With the changes, `bevy_ecs` took about half a second longer to compile.
```
rustc 1.71.0 (8ede3aae2 2023-07-12)
Host: x86_64-pc-windows-msvc
Target: x86_64-pc-windows-msvc
```